File: js/tests/data_structure_tests/disjoint_set.test.js
import {DisjointSet} from "../../data structures/DisjointSet.js";




test('test 1', () => {
    let obj1 = new DisjointSet(20);
    expect(obj1.disjointSet).toStrictEqual(Array(20).fill(-1));
});


test('test2', () => {
    let obj1 = new DisjointSet(20)
    obj1.union(4,5);
    obj1.union(4,9);
    obj1.union(4,15);
    obj1.union(4,19);

    expect(4).toBe(obj1.find(4));
    expect(4).toBe(obj1.find(9));
    expect(4).toBe(obj1.find(15));
    expect(4).toBe(obj1.find(19));

});

test('test3', () => {
    let obj1 = new DisjointSet(20);
    obj1.union(1,5);
    obj1.union(1,9);
    obj1.union(4,15);
    obj1.union(4,3);
    obj1.union(4,16);
    obj1.union(3,1);

    expect(4).toBe(obj1.find(4));
    expect(4).toBe(obj1.find(3)); 
    expect(4).toBe(obj1.find(1)); 
    expect(1).not.toBe(obj1.find(9)); 

});

File: py/test/data_structure_tests/DisjointSetTest.py
import unittest
from py.data_structures.DisjointSet import DisjointSet



class tests(unittest.TestCase):
    def test1(self):
        obj1 = DisjointSet(20)
        self.assertEqual([-1]*20, obj1.forest)

    def test2(self):
        obj1 = DisjointSet(20)
        obj1.union(4,5)
        obj1.union(4,9)
        obj1.union(4,15)
        obj1.union(4,19)

        self.assertEqual(4, obj1.find(4))
        self.assertEqual(4, obj1.find(9))
        self.assertEqual(4, obj1.find(15))
        self.assertEqual(4, obj1.find(19))

    def test3(self):
        obj1 = DisjointSet(20)
        obj1.union(4,5)
        obj1.union(4,9)
        obj1.union(4,15)
        obj1.union(4,19)
        obj1.union(1,19)
        obj1.union(3,19)

        self.assertEqual(4, obj1.find(4))
        self.assertEqual(4, obj1.find(3))




if __name__ == "__main__":
    unittest.main()